Automattic/sensei
# Fix/lm sticky sidebar

Fixes #5912 

### Changes proposed in this Pull Request

* Makes the sidebar for `modern` and `full-video` templates sticky.
* For `modern` template if featured video is taller than the sidebar, then it makes the sidebar the same height.
* If the sidebar is very tall and does not fit into the viewport vertically, then it scrolls up/down so users can reach all the sidebar content.
* Updates sidebar position when the viewport is resized.

### Testing instructions


* Try out different scenarios. Like shown in the video below.
* Make sure the sticky sidebar sticks when it makes sense.


### Screenshot / Video

#### Regular Sticky Sidebar

https://user-images.githubusercontent.com/2578542/207894569-5c147d3a-eddf-4d7c-9f8b-a00aa1b13936.mp4

#### Full Video Sticky Sidebar

https://user-images.githubusercontent.com/2578542/207894759-300a7c85-72eb-4815-ad4c-660d5908b52d.mp4

#### Height Overflow Sticky Sidebar

https://user-images.githubusercontent.com/2578542/207894836-41d3a143-ee58-49ca-9a4f-416ec77c8216.mp4

#### Video Height Sticky Sidebar

https://user-images.githubusercontent.com/2578542/207894975-ecaf92c0-012c-4d32-9706-80790512d748.mp4

#### Video Full Height Overflow Sticky Sidebar

https://user-images.githubusercontent.com/2578542/207895099-a98321b2-ffe7-46aa-95c7-5322e8d45fdf.mp4
